courses-rest/src/test/java/no/sikt/nva/fs/CoursesByInstitutionOfLoggedInUserHandlerTest.java

```diff
@@ -1,65 +1,277 @@
+        
+        assertEquals(HttpURLConnection.HTTP_OK, gatewayResponse.getStatusCode());
+        assertThat(gatewayResponse.getHeaders(), hasKey(CONTENT_TYPE));
+        assertThat(gatewayResponse.getHeaders(), hasKey(ACCESS_CONTROL_ALLOW_ORIGIN));
+        assertThat(gatewayResponse.getBodyObject(Course[].class), emptyArray());
+    }
+    
+    @Test
+    void shouldReturnCoursesWhenInstitutionHasFsAccessAfterSummer() throws IOException {
+        // prepare:
+        final InputStream input = createRequest(SUPPORTED_INSTITUTION_PATH);
+        
+        final String responseBody2022 = IoUtils.stringFromResources(Path.of("oslometUndervisningResponse2022.json"));
+        stubFor(get(urlPathEqualTo(TAUGHT_COURSES_URL_PATH))
+            .withHeader(AUTHORIZATION, equalTo(SUCCESS_AUTHORIZATION_HEADER_VALUE))
+            .withQueryParam(YEAR_QUERY_PARAM_NAME, equalTo(YEAR_2022))
+            .withQueryParam(DB_ID_QUERY_PARAM_NAME, equalTo("true"))
+            .withQueryParam(INSTITUTION_QUERY_PARAM_NAME, equalTo("215"))
+            .withQueryParam(LIMIT_QUERY_PARAM_NAME, equalTo("0"))
+            .willReturn(WireMock.ok()
+                .withHeader(CONTENT_TYPE, APPLICATION_JSON_CONTENT_TYPE_VALUE)
+                .withBody(responseBody2022)));
+        
+        final String responseBody2023 = IoUtils.stringFromResources(Path.of("oslometUndervisningResponse2023.json"));
+        stubFor(get(urlPathEqualTo(TAUGHT_COURSES_URL_PATH))
+            .withHeader(AUTHORIZATION, equalTo(SUCCESS_AUTHORIZATION_HEADER_VALUE))
+            .withQueryParam(YEAR_QUERY_PARAM_NAME, equalTo(YEAR_2023))
+            .withQueryParam(DB_ID_QUERY_PARAM_NAME, equalTo("true"))
+            .withQueryParam(INSTITUTION_QUERY_PARAM_NAME, equalTo("215"))
+            .withQueryParam(LIMIT_QUERY_PARAM_NAME, equalTo("0"))
+            .willReturn(WireMock.ok()
+                .withHeader(CONTENT_TYPE, APPLICATION_JSON_CONTENT_TYPE_VALUE)
+                .withBody(responseBody2023)));
+        
+        this.handler = new CoursesByInstitutionOfLoggedInUserHandler(environment, FIXED_CLOCK);
+        
+        // execute:
+        handler.handleRequest(input, output, context);
+        
+        // verify:
+        final GatewayResponse<Course[]> gatewayResponse = GatewayResponse.fromOutputStream(output, Course[].class);
+        
+        assertEquals(HttpURLConnection.HTTP_OK, gatewayResponse.getStatusCode());
+        assertThat(gatewayResponse.getHeaders(), hasKey(CONTENT_TYPE));
+        assertThat(gatewayResponse.getHeaders(), hasKey(ACCESS_CONTROL_ALLOW_ORIGIN));
+        
+        final Course[] courses = gatewayResponse.getBodyObject(Course[].class);
+        assertThat(courses, arrayWithSize(5));
+        assertThat(courses, arrayContaining(COURSE_A_AUTUMN_2022,
+            COURSE_AE_AUTUMN_2022,
+            COURSE_A_SPRING_2023,
+            COURSE_OE_SPRING_2023,
+            COURSE_AA_SPRING_2023));
```

Comment:
there is a hidden expectation here.

Reply:
Adjusted name of test method to include "Sorted".